controllers/nutanixcluster_controller.go

```diff
@@ -418,3 +418,10 @@ func (r *NutanixClusterReconciler) reconcileCredentialRef(ctx context.Context, n
 	}
 	return nil
 }
+
+func mustGetCredentialRefForCluster(nutanixCluster *infrav1.NutanixCluster) (*credentialTypes.NutanixCredentialReference, error) {
```

Comment:
nutanixCluster being nil and returning an error seemed like the controller's concern, added a helper function here.

Reply:
Not sure if it's just me but when I see `must` in a function name, I expect that function to panic instead of error. Given it's being used in just two places, we can just do a nil check in those places. Alternatively, we can consider renaming this function to something like `GetNutanixClusterCredentialRef`.